### PR TITLE
support image and resource patch when customJobYaml

### PR DESCRIPTION
**Pull request by tornado404**
### PR types
Function optimization

### PR changes
support image and resource patch when customJobYaml

### Describe
support image and resource patch when customJobYaml

### CODE REVIEW

File: pkg/job/runtime_v2/job/single/kube_single_job.go [hunk 1]
```diff
@@ -87,6 +87,16 @@ func (sp *KubeSingleJob) Submit(ctx context.Context, job *api.PFJob) error {
 		log.Errorf("build scheduling policy for %s failed, err: %v", sp.String(jobName), err)
 		return err
 	}
+	if singlePod == nil || job == nil {
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/job/single/kube_single_job.go]: Dead code.

File: pkg/job/runtime_v2/job/single/kube_single_job.go [hunk 2]
```diff
@@ -126,21 +136,41 @@ func (sp *KubeSingleJob) customSingleJob(jobPod *v1.Pod, job *api.PFJob) error {
 		return fmt.Errorf("jobSpec or PFJob is nil")
 	}
 	// TODO: add more patch
+
+	if err := sp.setImageAndResource(&jobPod.Spec, job.Tasks[0]); err != nil {
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/job/single/kube_single_job.go]: It might be better to split into `setImage` And `setResource`

Comment (tornado404) [reply pkg/job/runtime_v2/job/single/kube_single_job.go]: OK

File: pkg/job/runtime_v2/job/single/kube_single_job.go (hunk 1 above)
Comment (tornado404) [reply pkg/job/runtime_v2/job/single/kube_single_job.go]: removed

**Review by luoyuedong — APPROVED** [review]

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**